upgrade.sh

```diff
@@ -52,6 +52,66 @@ function unbound_psp_check() {
     echo "cray-unbound-coredns-psp check Done"
 }
 
+# Function to check cray-sysmgmt-health chart with app version 9.3.1 for prometheus-operator and retain old PVs data.
+function sysmgmt_health() {
+    echo "Checking for chart version of cray-sysmgmt-health"
+    version="45.1"
+    if [ $(helm ls -o json --namespace sysmgmt-health| jq -r --argjson version $version '.[] | select(.app_version | sub(".[0-9]$";"") | tonumber | . < $version).name') ]
+    then
+    prom_pvc="prometheus-cray-sysmgmt-health-promet-prometheus-db-prometheus-cray-sysmgmt-health-promet-prometheus-0"
+    alert_pvc="alertmanager-cray-sysmgmt-health-promet-alertmanager-db-alertmanager-cray-sysmgmt-health-promet-alertmanager-0"
+    echo "Get PV for both prometheus and Alertmanager"
+    prom_pv=$(kubectl get pvc -n sysmgmt-health --output=json $prom_pvc |jq '.spec."volumeName"')
```

Comment:
Best to not use jq for these outputs and just have kubectl do the work.  Won't tag all the spots but jq is entirely unnecessary for this output. e.g.:

```
ncn-m001:~ # kubectl get pvc -n sysmgmt-health -o jsonpath='{.spec.volumeName}' prometheus-cray-sysmgmt-health-promet-prometheus-db-prometheus-cray-sysmgmt-health-promet-prometheus-0
pvc-0a42ca5a-dfb1-44ff-b9bd-72080e25b5b7
```

Reply:
okay

Reply:
I'm still seeing jq calls in pipes, did a commit get missed perhaps?

Reply:
I can't see jq calls for the lines you mentioned. We have made changes.
    prom_pv=$(kubectl get pvc -n sysmgmt-health -o jsonpath='{.spec.volumeName}' $prom_pvc)
    alert_pv=$(kubectl get pvc -n sysmgmt-health -o jsonpath='{.spec.volumeName}' $alert_pvc)

https://github.com/Cray-HPE/csm/blob/5a7c87ddfbaaf5f8f2e6230b43d6b08bbe3b0938/upgrade.sh#L64

Reply:
made changes. 
`if [ ! -z  $(helm ls -o json --namespace sysmgmt-health| jq -r --argjson version $version '.[] | select(.app_version | sub(".[0-9]$";"") | tonumber | . < $version).name') ]  This if condition still has jq calls as it is required here.`